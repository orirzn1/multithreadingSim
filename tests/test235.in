L8
S6
O8
N7

T0
I@0x00000000
ADDI $0, $7, 0x2
LOAD $1, $1, $0
SUB $6, $6, $1
STORE $5, $4, $3
SUBI $4, $1, 0x4
ADD $7, $5, $3
ADDI $2, $3, 0x2
STORE $7, $5, $2
STORE $5, $3, 0x4
HALT $0

T1
I@0x00000000
ADDI $0, $5, 0x2
STORE $4, $2, 0x2
SUB $5, $1, $5
SUBI $4, $1, 0x1
LOAD $5, $1, 0x4
ADDI $4, $2, 0x1
SUBI $1, $5, 0x4
STORE $6, $1, 0x1
HALT $0

T2
I@0x00000000
STORE $7, $5, $6
SUBI $5, $0, 0x3
SUB $1, $0, $3
ADDI $5, $0, 0x3
LOAD $4, $4, $3
HALT $0

T3
I@0x00000000
ADDI $1, $6, 0x3
SUBI $6, $7, 0x2
ADD $5, $3, $6
SUBI $5, $0, 0x3
LOAD $5, $3, 0x1
ADDI $7, $0, 0x4
SUBI $5, $4, 0x3
STORE $2, $3, 0x2
HALT $0

T4
I@0x00000000
ADD $7, $2, $5
SUBI $6, $1, 0x4
SUBI $7, $7, 0x3
ADDI $2, $4, 0x3
LOAD $7, $7, 0x2
ADD $0, $1, $6
STORE $4, $7, 0x3
HALT $0

T5
I@0x00000000
SUBI $3, $1, 0x1
ADDI $2, $5, 0x1
SUBI $5, $2, 0x2
STORE $2, $4, 0x4
STORE $4, $6, 0x2
ADDI $6, $1, 0x3
HALT $0

T6
I@0x00000000
SUB $3, $2, $3
ADD $1, $3, $6
SUB $1, $6, $7
SUBI $4, $7, 0x4
ADD $5, $0, $3
HALT $0

D@0x0
0x8
0x5
0x6
0x9
0x6
0x7
0x9
0x9
