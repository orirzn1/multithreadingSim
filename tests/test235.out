
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x2	R1 = 0x8	R2 = 0x2	R3 = 0x0	R4 = 0x4	R5 = 0x0	R6 = 0xFFFFFFF8	R7 = 0x0
Register file thread id 1:
	R0 = 0x2	R1 = 0xFFFFFFFC	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x3	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFD	R6 = 0xFFFFFFFE	R7 = 0x4
Register file thread id 4:
	R0 = 0xFFFFFFFC	R1 = 0x0	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFC	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0xFFFFFFFF	R6 = 0x3	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0x0
Blocked MT CPI for this program 4.054545

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x2	R1 = 0x0	R2 = 0x2	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x2	R1 = 0xFFFFFFFC	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x3	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFD	R6 = 0xFFFFFFFE	R7 = 0x4
Register file thread id 4:
	R0 = 0xFFFFFFFC	R1 = 0x0	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFC	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0xFFFFFFFF	R6 = 0x3	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0x0
Finegrained Multithreading CPI for this program 1.090909

